src/phoebus_driver.cpp

```diff
@@ -1019,6 +1020,39 @@ TaskListStatus PhoebusDriver::MonteCarloStep() {
 void UserWorkBeforeOutput(MeshBlock *pmb, ParameterInput *pin) {
   auto tracer_pkg = pmb->packages.Get("tracers");
   bool do_tracers = tracer_pkg->Param<bool>("active");
+  bool is_stellarcollapse = (pin->GetString("eos", "type") == "StellarCollapse");
+
+  if (is_stellarcollapse) {
+    namespace p = fluid_prim;
+    auto rc = pmb->meshblock_data.Get().get();
+    PackIndexMap imap;
+    auto v = rc->PackVariables(
+        {p::density::name(), p::ye::name(), p::temperature::name(), p::entropy::name()},
+        imap);
+
+    const int irho = imap[fluid_prim::density::name()].first;
+    const int iye = imap[fluid_prim::ye::name()].first;
+    const int itemp = imap[fluid_prim::temperature::name()].first;
+    const int is = imap[fluid_prim::entropy::name()].first;
+
+    IndexRange ib = rc->GetBoundsI(IndexDomain::interior);
+    IndexRange jb = rc->GetBoundsJ(IndexDomain::interior);
+    IndexRange kb = rc->GetBoundsK(IndexDomain::interior);
+
+    auto eos = pmb->packages.Get("eos")->Param<Microphysics::EOS::EOS>("d.EOS");
+    singularity::StellarCollapse eos_sc =
+        eos.GetUnmodifiedObject().get<singularity::StellarCollapse>();
+    parthenon::par_for(
+        DEFAULT_LOOP_PATTERN, "CoolingFunctionCalculateFourForce", DevExecSpace(), kb.s,
```

Comment:
Should change loop label, e.g., `UserWorkBeforeOutput::Entropy` or something.
Also, is it necessary to specify the execution space? cc: @Yurlungur

Reply:
I *think* the default execution space, which will be device, is fine.